### PR TITLE
Fix CRD validation error on 'items' field

### LINKED ISSUE #68466: CRD cannot have a field named 'items'
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
I cannot have a resource which has a field named ```items``` it doesn't pass validation.
As an example I will modify sample-controller (https://github.com/kubernetes/sample-controller) and add ```items``` to ```artifacts/examples/example-foo.yaml```:

```
apiVersion: samplecontroller.k8s.io/v1alpha1
kind: Foo
metadata:
  name: example-foo
spec:
  deploymentName: example-foo
  replicas: 1
  items: # <-- cannot be set
  - "foo"
  - "bar"
```

after executing:

```kubectl create -f artifacts/examples/example-foo.yaml```

I receive an error:

```
The Foo "example-foo" is invalid: []: Invalid value: map[string]interface {}{"apiVersion":"samplecontroller.k8s.io/v1alpha1", "kind":"Foo", "metadata":map[string]interface {}{"name":"example-foo", "namespace":"default", "generation":1, "creationTimestamp":"2018-09-10T10:31:05Z", "uid":"9f41ebca-b4e4-11e8-ab9b-525500d15501", "selfLink":"", "clusterName":""}, "spec":map[string]interface {}{"deploymentName":"example-foo", "items":[]interface {}{"foo", "bar"}, "replicas":1}}: validation failure list:
type in spec is required
```

Validation expects that I will set a ```type``` field as well, spec below doesn't return the error:
```
apiVersion: samplecontroller.k8s.io/v1alpha1
kind: Foo
metadata:
  name: example-foo
spec:
  deploymentName: example-foo
  replicas: 1
  items:
  - "foo"
  - "bar"
  type: array  # <-- with 'type' it works
```

After all, in my opinion, it is a bug. I think this connected to OpenApi schema validation, that ```items``` keyword should always be in pair with ```type```.

**What you expected to happen**:
I would expect that I am able to use a field named ```items``` in CRD spec and it passes validation.
Similar to ConfigMap, it has an ```items``` field, https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#add-configmap-data-to-a-specific-path-in-the-volume.

**How to reproduce it (as minimally and precisely as possible)**:
The easiest way is to use https://github.com/kubernetes/sample-controller,
change ```artifacts/examples/example-foo.yaml```
from:
```
apiVersion: samplecontroller.k8s.io/v1alpha1
kind: Foo
metadata:
  name: example-foo
spec:
  deploymentName: example-foo
  replicas: 1
```
to:
```
apiVersion: samplecontroller.k8s.io/v1alpha1
kind: Foo
metadata:
  name: example-foo
spec:
  deploymentName: example-foo
  replicas: 1
  items:
  - "foo"
  - "bar"
```
then execute:

```kubectl -f create artifacts/examples/crd-validation.yaml``` (it doesn't matter if you will change something here)

```kubectl -f create artifacts/examples/example-foo.yaml```

After mentioned steps, you should receive the error (btw you don't have to change the ```Foo``` structure to reproduce this)

**Anything else we need to know?**:
No.

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:13:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:00:59Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Tested as well with k8s-1.11.0, the same problem occurred.
- OS (e.g. from /etc/os-release): Fedora27
- Kernel (e.g. `uname -a`) :4.17.19-100.fc27.x86_64

### PR DESCRIPTION
**Pull request by tossmilestone**
Signed-off-by: He Xiaoxi <tossmilestone@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

 /kind bug

> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Fix CRD validation errror on 'items' field

**Which issue(s) this PR fixes**:

Fixes #68466

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix CRD validation error on 'items' field.
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -140,26 +140,26 @@ github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8 h1:DujepqpGd1hyOd7a
...
+github.com/go-openapi/runtime v0.19.0 h1:sU6pp4dSV2sGlNKKyHxZzi1m1kG4WnYtWcJ+HYbygjE=
+github.com/go-openapi/runtime v0.19.0/go.mod h1:OwNfisksmmaZse4+gpV3Ne9AyMOlP1lt4sK4FXt0O64=
+github.com/go-openapi/spec v0.19.0 h1:A4SZ6IWh3lnjH0rG0Z5lkxazMGBECtrZcbyYQi+64k4=
+github.com/go-openapi/spec v0.19.0/go.mod h1:XkF/MOi14NmjsfZ8VtAKf8pIlbZzyoTvZsdfssdxcBI=
+github.com/go-openapi/strfmt v0.19.0 h1:0Dn9qy1G9+UJfRU7TR8bmdGxb4uifB7HNrJjOnV0yPk=
+github.com/go-openapi/strfmt v0.19.0/go.mod h1:+uW+93UVvGGq2qGaZxdDeJqSAqBqBdl+ZPMF/cC8nDY=
+github.com/go-openapi/swag v0.19.0 h1:Kg7Wl7LkTPlmc393QZQ/5rQadPhi7pBVEMZxyTi0Ii8=
+github.com/go-openapi/swag v0.19.0/go.mod h1:AByQ+nYG6gQg71GINrmuDXCPWdL640yX49/kXLo40Tg=
+github.com/go-openapi/validate v0.19.1-0.20190403094941-3fd099a3cf2e h1:G1v+JbiQPzgvfzXbfy553wxFIE374JXjCQXBggouHTE=
+github.com/go-openapi/validate v0.19.1-0.20190403094941-3fd099a3cf2e/go.mod h1:Uh4HdOzKt19xGIGm1qHf/ofbX1YQ4Y+MYsct2VUrAJ4=
```
Comment (liggitt) [inline go.sum]: is there a tagged version of `github.com/go-openapi/validate` that contains the fix? if not, do we know when that could be available? we'd prefer not to move from a tag to a random commit

File: go.mod [hunk 2]
```diff
@@ -11,129 +11,127 @@ require (
 	cloud.google.com/go v0.0.0-20160913182117-3b1ae45394a2
 	github.com/Azure/azure-sdk-for-go v21.3.0+incompatible
 	github.com/Azure/go-autorest v11.1.0+incompatible
-	github.com/GeertJohan/go.rice v0.0.0-20170420135705-c02ca9a983da // indirect
+	github.com/GeertJohan/go.rice v0.0.0-00010101000000-000000000000 // indirect
```
Comment (liggitt) [inline go.mod]: these versionless require directives aren't right... can you revert the `require` bits to this file (your new `replace` directives are fine) and rerun `hack/update-vendor.sh`?

File: go.sum (hunk 1 above)
Comment (tossmilestone) [reply go.sum]: I have asked the maintainer to add a tag, will resolve it soon.

File: go.mod (hunk 2 above)
Comment (tossmilestone) [reply go.mod]: ok

**Comment by tossmilestone** [comment]
Suspended for the next release of `go-openapi/validate`

**Comment by liggitt** [comment]
looks like v0.19.2 was tagged with this fix

**Comment by tossmilestone** [comment]
Yes, it's time to merge the PR.

**Comment by tossmilestone** [comment]
@liggitt this PR is done now, PTAL.

**Comment by liggitt** [comment]
thanks for the quick update. the dependencies you pulled in have transitive dependencies we need to also bump. `hack/lint-dependencies.sh | grep -v golang` will show the ones we need to bump. Go ahead and put all the openapi ones in a single commit, and the others in their own commits.

**Comment by tossmilestone** [comment]
@liggitt all bump should be done, please check again.

**Comment by liggitt** [comment]
@tossmilestone, thanks for all the work on this

**Comment by cliveseldon** [comment]
Unfortunately this fix has not solved the core issue. See https://github.com/go-openapi/validate/issues/108
This has gone into k8s 1.16 and causes the k8sapi-server to panic. I will try to raise a bug issue for this to fully document. @liggitt

**Comment by cliveseldon** [comment]
On further investigation I think this is due to `x-kubernetes-int-or-string` 

See https://github.com/kubernetes/kubernetes/issues/83778

**Pull request merged**